code-studio/src/js/initApp/initApp.js

```diff
@@ -169,8 +169,8 @@ window.apps = {
         // If we're readOnly, source hasn't changed at all
         source = Blockly.mainBlockSpace.isReadOnly() ? currentLevelSource :
           Blockly.Xml.domToText(Blockly.Xml.blockSpaceToDom(Blockly.mainBlockSpace));
-      } else {
-        source = window.Applab && Applab.getCode();
+      } else if (appOptions.getCode) {
+        source = appOptions.getCode();
```

Comment:
Unrelated to this PR: Did building `code-studio` to test this stuff give you any trouble?